https/gw2Accounts.js

```diff
@@ -61,23 +61,32 @@ const assignedGiftees = functions.https.onCall(async ({user}, context) => {
   if(gifter_uuid.error){return {error: "no API key set"}}
   gifter_uuid = gifter_uuid.success
 
-  let giftee = await db.collection('events').doc(YEAR).collection('participants').where('gifter', '==', gifter_uuid).get()
+  let giftee = await db.collection('events').doc(EVENT).collection('participants').where('gifter', '==', gifter_uuid).get()
   if (giftee.empty) {return {error: "No valid users"}}
 
   let gifteeArrayRaw = []
   giftee.forEach(doc => {gifteeArrayRaw.push(doc.data())});
 
-  // array in case the user is sending gifts to f2p folks
+  // array in case the user is sending gifts to multiple folks
   let gifteeArray = []
-  for (const gifteeData of gifteeArrayRaw) {
-    // eslint-disable-next-line no-await-in-loop
+
+  let promises = gifteeArrayRaw.map(async (gifteeData) => {
```

Comment:
Shouldn't this be a `forEach` instead? `map` loops through and expects a return statement, while here it seems like we only want to loop through.

Reply:
forEach preforms a function for each tiem in teh array, however I am using map to return an array of promises which all have to be resolved to move on, basically allowing them to run in parallell instead of series

Reply:
I didn't see the return statement, so that's on me for misreading.

Reply:
Ye the map is somewhat streange in teh way it handles the promises, works pretty well though